src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftImageTagger.java

```diff
@@ -17,272 +17,274 @@
-					} catch (com.openshift.restclient.OpenShiftException e) {
-						String createJson = "{\"kind\": \"ImageStream\",\"apiVersion\": \"v1\",\"metadata\": {\"name\": \"" +
-								destStream + "\",\"creationTimestamp\": null},\"spec\": {},\"status\": {\"dockerImageRepository\": \"\"}}";
-
-						Map<String,String> newOverrides = new HashMap<String,String>(overrides);
-						// we don't want this step's source namespace to be the creator's namespace,
-						// but rather this step's destination namespace, so clear out the override
-						// and then reuse all other overrides.
-						newOverrides.remove("namespace");
-
-						String token = destToken;
-						if ( token == null || token.isEmpty() ) {
-							token = getAuthToken();
-						}
-
-						OpenShiftCreator isCreator = new OpenShiftCreator(getApiURL(newOverrides), destNS, token, ""+chatty, createJson);
-						isCreator.setAuth(Auth.createInstance(chatty ? listener : null, getApiURL(newOverrides), overrides));
-
-						boolean newISCreated = isCreator.coreLogic(launcher, listener, newOverrides);
-
-						if (!newISCreated) {
-							listener.getLogger().println(String.format(MessageConstants.EXIT_TAG_CANNOT_CREATE_DEST_IS, destStream, destNS));
-							return false;
-						}
-
-						destIS = srcClient.get(ResourceKind.IMAGE_STREAM, destStream, destNS);
-					}
-
-					if (destIS == null) {
-						listener.getLogger().println(String.format(MessageConstants.EXIT_TAG_CANNOT_GET_IS, destStream, destNS));
-						return false;
-					}
-
-					destIS.addTag( destTag, tagType, srcImageID, srcNS);
-					if (chatty) {
-						listener.getLogger().println("\n updated image stream json " + ((ImageStream) destIS).getNode().toJSONString(false) + " with tag: " + destTag );
-					}
-
-					try {
-						destClient.update(destIS);
-						break;
-					} catch ( OpenShiftException ose ) {
-						if ( ose.getStatus().getCode() == 403 && retries > 1 ) { // If resource was updating, retry
-							continue;
-						}
-						throw ose;
-					}
-				}
-			}
-
-	    	listener.getLogger().println(String.format(MessageConstants.EXIT_OK, DISPLAY_NAME));
-			return true;
-    	} else {
-    		return false;
-    	}
-
-	}
+    final static String DISPLAY_NAME = "Tag OpenShift Image";
+
+    default String getDisplayName() {
+        return DISPLAY_NAME;
+    }
+
+    String getAlias();
+
+    String getSrcTag();
+
+    String getDestTag();
+
+    String getSrcStream();
+
+    String getDestStream();
+
+    String getDestinationNamespace();
+
+    String getDestinationAuthToken();
+
+    default String getAlias(Map<String, String> overrides) {
+        return getOverride(getAlias(), overrides);
+    }
+
+    default String getSrcTag(Map<String, String> overrides) {
+        String val = getOverride(getSrcTag(), overrides);
+        // If there is no source namespace, default to jenkins namespace
+        if (val.length() == 0 && overrides != null && overrides.containsKey(NAMESPACE_ENV_VAR)) {
```

Comment:
I think this was meant for getDestinationNamespace .... that tag should not be derived from the namespace env var ... that said, we can just leverage getNamespace there